test/SILOptimizer/sil_combine.sil

```diff
@@ -4226,3 +4226,39 @@ bb0:
   return %16 : $Int
 } // end sil function 'test_global_value'
 
+final class Kp1 {
+  @_hasStorage @_hasInitialValue final var b: Kp2? { get set }
+  @objc deinit
+  init()
+}
+
+class Kp2 {
+  @_hasStorage @_hasInitialValue var s: String { get set }
+  @objc deinit
+  init()
+}
+
+sil @kpgetter : $@convention(thin) (@in_guaranteed Kp2) -> @out String
+sil @kpsetter : $@convention(thin) (@in_guaranteed String, @in_guaranteed Kp2) -> ()
+sil @swift_getAtKeyPath : $@convention(thin) <τ_0_0, τ_0_1> (@in_guaranteed τ_0_0, @guaranteed KeyPath<τ_0_0, τ_0_1>) -> @out τ_0_1
+
+// CHECK-LABEL: sil @test_optional_chaining_keypath
```

Comment:
I would never guess what that test is supposed to do. You can just paste the relevant source lines from the bug and explain how SILCombine lowers this optional keypath.

Reply:
I added a comment